File: VB/DateRangeFilterViaXpoDataSource/Northwind.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports DevExpress.Xpo
Namespace Northwind

	<Persistent("Customers")> _
	Public Class Customer
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private customerID_Renamed As String

		<Key> _
		Public Property CustomerID() As String
			Get
				Return customerID_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CustomerID", customerID_Renamed, value)
			End Set
		End Property

		Private contactTitle_Renamed As String
		Public Property ContactTitle() As String
			Get
				Return contactTitle_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("ContactTitle", contactTitle_Renamed, value)
			End Set
		End Property

		Private companyName_Renamed As String
		Public Property CompanyName() As String
			Get
				Return companyName_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("CompanyName", companyName_Renamed, value)
			End Set
		End Property

	Private address_Renamed As String
		Public Property Address() As String
			Get
				Return address_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Address", address_Renamed, value)
			End Set
		End Property

		<Association("CustomerOrders", GetType(Order))> _
		Public ReadOnly Property Orders() As XPCollection(Of Order)
			Get
				Return GetCollection(Of Order)("Orders")
			End Get
		End Property
	End Class

	<Persistent("Orders")> _
	Public Class Order
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Key(True), Persistent> _
		Private OrderID As Integer

		<PersistentAlias("OrderID")> _
		Public ReadOnly Property ID() As Integer
			Get
				Return OrderID
			End Get
		End Property

		Private shippedDate_Renamed As DateTime
		Public Property ShippedDate() As DateTime
			Get
				Return shippedDate_Renamed
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("ShippedDate", shippedDate_Renamed, value)
			End Set
		End Property

		Private customer_Renamed As Customer
		<Persistent("CustomerID"), Association("CustomerOrders")> _
		Public Property Customer() As Customer
			Get
				Return customer_Renamed
			End Get
			Set(ByVal value As Customer)
				SetPropertyValue("Customer", customer_Renamed, value)
			End Set
		End Property
	End Class

	<Persistent("Employees")> _
	Public Class Employee
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		<Key(True), Persistent> _
		Public EmployeeID As Integer

		Private _FirstName As String
		Public Property FirstName() As String
			Get
				Return _FirstName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("FirstName", _FirstName, value)
			End Set
		End Property
		Private _LastName As String
		Public Property LastName() As String
			Get
				Return _LastName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("LastName", _LastName, value)
			End Set
		End Property

		<PersistentAlias("Concat([FirstName], ' ', [LastName])")> _
		Public ReadOnly Property FullName() As String
			Get
				Return CStr(EvaluateAlias("FullName"))
			End Get
		End Property
	End Class
End Namespace

File: CS/DateRangeFilterViaXpoDataSource/Northwind.cs
namespace Northwind {
    using System;
    using DevExpress.Xpo;

    [Persistent("Customers")]
    public class Customer : XPLiteObject {
        public Customer(Session session) : base(session) { }

        string customerID;
        
        [Key]
        public string CustomerID {
            get { return customerID; }
            set { SetPropertyValue("CustomerID", ref customerID, value); }
        }

        string contactTitle;
        public string ContactTitle {
            get { return contactTitle; }
            set { SetPropertyValue("ContactTitle", ref contactTitle, value); }
        }

        string companyName;
        public string CompanyName {
            get { return companyName; }
            set { SetPropertyValue("CompanyName", ref companyName, value); }
        }

	string address;
        public string Address {
            get { return address; }
            set { SetPropertyValue("Address", ref address, value); }
        }        

        [Association("CustomerOrders", typeof(Order))]
        public XPCollection<Order> Orders {
            get {
                return GetCollection<Order>("Orders");
            }
        }
    }

    [Persistent("Orders")]
    public class Order : XPLiteObject {
        public Order(Session session) : base(session) { }

        [Key(true), Persistent]
        int OrderID;

        [PersistentAlias("OrderID")]
        public int ID {
            get { return OrderID; }
        }

        DateTime shippedDate;
        public DateTime ShippedDate {
            get { return shippedDate; }
            set { SetPropertyValue("ShippedDate", ref shippedDate, value); }
        }

        Customer customer;
        [Persistent("CustomerID"), Association("CustomerOrders")]
        public Customer Customer {
            get { return customer; }
            set { SetPropertyValue("Customer", ref customer, value); }
        }
    }

    [Persistent("Employees")]
    public class Employee : XPLiteObject {
        public Employee(Session session) : base(session) { }

        [Key(true), Persistent]
        public int EmployeeID;

        private string _FirstName;
        public string FirstName {
            get {
                return _FirstName;
            }
            set {
                SetPropertyValue("FirstName", ref _FirstName, value);
            }
        }
        private string _LastName;
        public string LastName {
            get {
                return _LastName;
            }
            set {
                SetPropertyValue("LastName", ref _LastName, value);
            }
        }

        [PersistentAlias("Concat([FirstName], ' ', [LastName])")]
        public string FullName {
            get {
                return (string)EvaluateAlias("FullName");
            }
        }
    }
}
